File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
// 단, 하나의 지표에서 각 성격 유형 점수가 같으면, 
  // 두 성격 유형 중 사전 순으로 빠른 성격 유형을 검사자의 성격 유형이라고 판단
// 1~3 -> survey[i]의 첫 요소, 5~7 -> survey[i]의 두번째 요소, 4는 continue

// 반복문 돌면서 1~4지표의 각 점수 합산
// 최종적으로 1지표부터 성격유형 하나씩 찍어주기   [[R,T], [C,F], [J,M], [A,N]]

function solution(survey, choices) {
  let answer = '';
  let obj = {'R':0, 'T':0, 'C':0, 'F':0, 'J':0, 'M':0, 'A':0, 'N':0}

  for (let i=0; i<survey.length; i++) {
    if (choices[i] >= 5) obj[survey[i][1]] += (choices[i]-4)
    else if (choices[i] <= 3) obj[survey[i][0]] += (4-choices[i])
    else continue
  }

  if (obj['R'] >= obj['T']) answer += 'R' // 같은 경우 사전순까지 걍 러프하게 해줌
  else answer += 'T'

  if (obj['C'] >= obj['F']) answer += 'C'
  else answer += 'F'

  if (obj['J'] >= obj['M']) answer += 'J'
  else answer += 'M'

  if (obj['A'] >= obj['N']) answer += 'A'
  else answer += 'N'
  
  return answer;
}


console.log(solution(["AN", "CF", "MJ", "RT", "NA"], [5, 3, 2, 7, 5])) // TCMA
console.log(solution(["TR", "RT", "TR"],[7, 1, 3])) // RCJA

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
# 1트 - 실패(7/20)
    # survey와 choices를 같이 순회하면서 점수 표를 만들고,
    # 비교하면서 유형 return (점수 같으면 사전 순)
    # [R T] / [C F] / [J M] / [A N]
    # 1 2 3 4 5 6 7 
    # 앞에것 3 - 2 - 1
    # 뒤에것 1 - 2 - 3
    # 4점은 continue
# ? 어 근데, 같은 유형의 점수가 2번 나오면? 덮어씌우면 되나 그냥?
    # '검사 결과는 모든 질문의 성격 유형 점수를 더하여' 문제에 나와있었음;;
    # 이거 반영해서 코드수정ㄱㄱ
def solution(survey, choices):
    answer = ''
    # 아래처럼 초기값 넣어줘야 survey반복문 돌릴 때, 존재않는 key에러가 안남
    dict = {'R': 0, 'T': 0, 'C': 0, 'F': 0, 'J': 0, 'M': 0, 'A': 0, 'N': 0,}
    # choice[i]의 점수가 5이상이면 뒤의 문자 / 3이하면 앞의 문자점수
    score = [3,2,1, 0, 1,2,3]

    for k in range(len(survey)):
        if choices[k] >= 5:
            dict[survey[k][1]] += score[choices[k]-1]
        elif choices[k] <= 3:
            dict[survey[k][0]] += score[choices[k]-1]

    # =조건 넣으면, 사전순도 같이 처리되도록ㅇㅇ.
    answer += 'R' if dict['R'] >= dict['T'] else 'T'
    answer += 'C' if dict['C'] >= dict['F'] else 'F'
    answer += 'J' if dict['J'] >= dict['M'] else 'M'
    answer += 'A' if dict['A'] >= dict['N'] else 'N'
    
    return answer

print(solution(["AN", "CF", "MJ", "RT", "NA"],[5, 3, 2, 7, 5])) # "TCMA"
print(solution(["TR", "RT", "TR"],[7, 1, 3])) # "RCJA"